pkg/login/social/generic_oauth.go

```diff
@@ -209,6 +213,43 @@ func (s *SocialGenericOAuth) extractFromToken(token *oauth2.Token) *UserInfoJson
 		return nil
 	}
 
+	var headerBytes []byte
+	headerBytes, err = base64.RawURLEncoding.DecodeString(matched[1])
+	if err != nil {
+		s.log.Error("Error base64 decoding header", "header", matched[1], "error", err)
+		return nil
+	}
+
+	var header map[string]string
+	err = json.Unmarshal(headerBytes, &header)
+	if compression, ok := header["zip"]; ok {
+		switch strings.ToLower(compression) {
```

Comment:
```suggestion
		switch compression {
```

> The "zip" value is a case-sensitive string.

from https://tools.ietf.org/html/rfc7516#section-4.1.3

(and adjust the cases accordingly)

Reply:
Thanks.  Good catch - fixing